cmd/collector/app/collector.go

```diff
@@ -169,7 +177,18 @@ func (c *Collector) Close() error {
 		defer cancel()
 	}
 
-	// by now, we shouldn't have any in-flight requests anymore
+	// by now, we shouldn't have any in-flight requests anymore, close the processor and handlers
+	for _, closer := range []io.Closer{c.spanProcessor, c.zipkinSpansHandler, c.jaegerBatchesHandler, c.grpcHandler} {
```

Comment:
In all three span handlers the Close is no-op. What's the point of adding it?

Reply:
While the case is strong for removing it, as we have three implementations and none needs to close anything, this seems like the type of thing that would require a closer in some implementations. 

I can remove it, but the message that it sends is stronger than the fact that it's currently no-op.

Reply:
The message is sends is that these handlers might be stateful in some way and require Close to clean-up, which is kind of the opposite of what they are now: stateless proxies that simply forward to collector (while not managing the life cycle of collector proxy). I am in favor of simplicity, rather than adding code that might possibly be needed in the future.

It also complicates our internal build since now we also have to call close on these handlers, to respect their API.